src/lib.rs

```diff
@@ -74,19 +76,36 @@ struct SandboxFS {
 
 impl SandboxFS {
     /// Creates a new `SandboxFS` instance.
-    fn new() -> SandboxFS {
+    fn new(mappings: &Vec<Mapping>) -> io::Result<SandboxFS> {
         let root = {
-            let now = time::get_time();
-            let uid = unsafe { libc::getuid() } as u32;
-            let gid = unsafe { libc::getgid() } as u32;
-            nodes::Dir::new_root(now, uid, gid)
+            if mappings.len() == 0 {
+                let now = time::get_time();
+                let uid = unsafe { libc::getuid() } as u32;
+                let gid = unsafe { libc::getgid() } as u32;
+                nodes::Dir::new_root(now, uid, gid)
+            } else if mappings.len() == 1 {
+                if mappings[0].path != Path::new("/") {
+                    panic!("Unimplemented; only support a single mapping for the root directory");
+                }
+                let fs_attr = fs::symlink_metadata(&mappings[0].underlying_path)?;
+                if !fs_attr.is_dir() {
+                    warn!("Path {:?} is not a directory; got {:?}", &mappings[0].underlying_path,
+                        &fs_attr);
+                    return Err(io::Error::from_raw_os_error(libc::EIO));
```

Comment:
This whole `SandboxFS::new` method is still a hack to cope with the incomplete implementation of mappings. Once I implement the handling for all mappings (which first needs full node implementations), this will have to change. Wouldn't worry too much yet...